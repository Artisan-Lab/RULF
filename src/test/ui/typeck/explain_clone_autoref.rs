struct NotClone;

fn main() {
    clone_thing(&NotClone);
}

fn clone_thing(nc: &NotClone) -> NotClone {
    //~^ NOTE expected `NotClone` because of return type
    nc.clone()
    //~^ ERROR mismatched type
    //~| NOTE `NotClone` does not implement `Clone`, so `&NotClone` was cloned instead
    //~| NOTE expected struct `NotClone`, found `&NotClone`
}
