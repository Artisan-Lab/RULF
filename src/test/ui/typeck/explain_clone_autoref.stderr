error[E0308]: mismatched types
  --> $DIR/explain_clone_autoref.rs:9:5
   |
LL | fn clone_thing(nc: &NotClone) -> NotClone {
   |                                  -------- expected `NotClone` because of return type
LL |
LL |     nc.clone()
   |     ^^^^^^^^^^ expected struct `NotClone`, found `&NotClone`
   |
note: `NotClone` does not implement `Clone`, so `&NotClone` was cloned instead
  --> $DIR/explain_clone_autoref.rs:9:5
   |
LL |     nc.clone()
   |     ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
